### ISSUE TITLE
Zeitwerk eager load fails trying to load a helper that doesn't exist

### ISSUE THREAD

**Issue by tdeck** [fxn/zeitwerk#108]
Here's a really strange behavior I'm noticing. I have a small Rails application with a bunch of controllers that lack helpers (I generally delete the helper file if there's nothing in it). That's been working fine.

Now I created an Admin namespace with its own controller, and decided to write a test to ensure that every controller in that namespace descends from the admin controller, so my test calls `Zeitwerk::Loader.eager_load_all`.

This call works fine in the Rails console, but fails when it encounters a controller without a corresponding helper I get this error:

```
LoadError: cannot load such file -- /home/troy/.backed_up/projects/ruby/cmatch/app/helpers/admin/sessions_helper.rb
    app/controllers/admin/sessions_controller.rb:1:in `<top (required)>'
```
The controller in question doesn't explicitly require the helper (which doesn't exist). Moreover:

1. This happens with any controllers in my nested `Admin` namespace unless I pop in a dummy helper file for that controller
2. This happens with no other controllers (Admin is my only namespace)
3. It doesn't matter what the controller is, only that it inherits (perhaps indirectly) from `ActionController::Base`
4. I don't get this error when eager loading in the console using `Zeitwerk::Loader.eager_load_all`, only in tests

I realize this obscure scenario might not be a priority but thought it worth reporting. I'm using v2.2.2 in Rails 6.0,2

**Comment by fxn** [comment]
That is very strange, because Zeitwerk only eager loads what's in the file system. If someone is trying to load a file that does not exist, that is not Zeitwerk. I suspect it is not related to Zeitwerk itself.

Do you have a minimal way to reproduce?

**Comment by tdeck** [comment]
I'll try to get one to you. I've discovered it seems to depend on whether I run the tests ("bundle exec rake test" works but bin/rake test, bin/rails test, and "bundle exec rails test" fail), so it's probably just something strange in my environment that I haven't tracked down yet. I've grepped my whole project dir for references to that file and not found it, but I suspect you're right that it's unrelated to Zeitwerk.
